woocommerce/woocommerce-gutenberg-products-block
# Product Collection: Make it compatible with filter blocks

This PR introduces following changes:
1. All filter blocks now work with Product Collection block i.e. price filter, attributes filter, stock status filter, and rating filter blocks.
2. **Renaming and extending existing methods**: Some methods are renamed for better clarity, for example `build_query` to `build_frontend_query` and `update_rest_query` to `update_rest_query_in_editor`. 

Fixes #9876 

### Testing

#### Test that filter blocks work with the Product Collection block

1. Create a new post and add a 'Product Collection' block to it. 
2. Add all the filter blocks to the post. Alternatively, you can use the 'Product Filters' pattern to add all filter blocks swiftly. 
3. Save the post and navigate to the frontend. 
4. Confirm that all filters function in harmony with the 'Product Collection' block.

Here is a quick demo:

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/e4215e0f-d7bb-4039-8cff-4187ef809096

#### Test with Product Catalog template

1. Open the product catalog template within the editor.
2. Insert a 'Product Collection' block into the Product Catalog. When the 'Product Collection' block is added to the Product Catalog template, the 'Inherit query from template' option will be enabled by default. Keep it enabled. 
3. Save the changes and navigate to '/shop'.
4. Confirm that all product filter blocks operate as expected.

Here is a quick demo:

https://github.com/woocommerce/woocommerce-blocks/assets/16707866/6caa9b83-6093-471e-82a7-f14ed1416bbd


### WooCommerce Visibility




### Changelog

> Product Collection: Make it compatible with filter blocks
